js/models/messages.js

```diff
@@ -518,7 +518,15 @@
                                     catch (e) {
                                         handleError(e);
                                     }
-                                }, handleError);
+                                }, function(error) {
+                                    console.log('handleDataMessage: Message', message.idForLogging(), 'was deleted');
+
+                                    if (confirm) {
```

Comment:
I just pushed a commit to simplify this - no more `if (confirm)` checks but we do have an extra `try`/`catch` in case something goes wrong in error/delete handler: https://github.com/WhisperSystems/Signal-Desktop/pull/1346/commits/42c75813b546ebb51391f9909ab9079c4f918e06

One of the reasons things are kinda messed up in this method is that we're dealing with a lot of fake promises, that don't properly propagate errors. All of those `.save()` and `.fetch()` calls you see there are not real promises. Tricky stuff.